src/sqlitetablemodel.cpp

```diff
@@ -605,8 +605,10 @@ QString SqliteTableModel::customQuery(bool withRowid)
 
         for(QMap<int, QString>::const_iterator i=m_mWhere.constBegin();i!=m_mWhere.constEnd();++i)
         {
-            QString column = sqlb::escapeIdentifier(m_headers.at(i.key()));
-            where.append(QString("%1 %2 AND ").arg(column).arg(i.value()));
+            QString columnId = sqlb::escapeIdentifier(m_headers.at(i.key()));
+            if(m_vDisplayFormat.size() && m_vDisplayFormat.at(i.key()-1) != columnId)
+                columnId = sqlb::escapeIdentifier(m_headers.at(i.key()) + "_");
```

Comment:
But that breaks for tables like these:
```sql
CREATE TABLE `test` (
	`field`	INTEGER,
	`field_`	INTEGER
);
```

On the other hand, the same was true for the old code, just with different column names. So I think we can leave it like this unless you have an idea how to fix this easily 😄

Reply:
Well, in fact I started adding "_" to all columns and that is the safe approach. But it was so ugly :smile: I think, I'll go back to that. Better ugly than having broken combinations.

Reply:
I'm surprised. sqlite is still filtering by the wrong column when all fields have been renamed. For your example table:
  SELECT `_rowid_`,upper(`field`) AS `field_`,`field_` AS `field__` FROM `main`.`test` WHERE `field_` = 'hola'  ORDER BY `_rowid_` ASC LIMIT 0, 50000;
That filters by `field__` (original `field_`) and not by the new `field_`. If sqlite is going to always give precedence to the original columns, I don't see a efficient and simple way to solve the problem. As you said, the original implementation had the same problem when there were original columns named col1, col2, etc.

So I'll merge from my last commit :smile:

Reply:
Yeah, I think the only way to do this properly is to apply the display format conversion to the column and to the WHERE clause:
```sql
SELECT DATETIME(field) AS field, field_ FROM test WHERE DATETIME(field) LIKE '%2017%';
```

Reply:
Yes, that is the standard and safe solution. What I have read is that any WHERE applied to the aliases is actually disallowed by the SQL standard. I've implemented your idea and works well. Do you think the display format would be calculated twice or it would be optimized? In any case I think it's the best solution.

https://stackoverflow.com/questions/8370114/referring-to-a-column-alias-in-a-where-clause
https://dev.mysql.com/doc/refman/5.5/en/problems-with-alias.html

Reply:
Looks like it is calculated twice. But whatever, implementing this correctly seems more important to me than the slight performance benefit :smile:


```sql
CREATE TABLE `test` (
	`field`	INTEGER,
	`field_`	INTEGER
);
EXPLAIN SELECT DATETIME(field) AS field, field_ FROM test WHERE DATETIME(field) LIKE '%2017%';
```
prints

addr | opcode | p1 | p2 | p3 | p4 | p5 | comment
---- | ------ | -- | -- | -- | -- | -- | -------
0 | Init | 0 | 13 | 0 |  | 00 | 
1 | OpenRead | 0 | 2121 | 0 | 2 | 00 | 
2 | Rewind | 0 | 12 | 0 |  | 00 | 
3 | Column | 0 | 0 | 4 |  | 00 | 
4 | Function0 | 0 | 4 | 3 | **datetime(-1)** | 01 | 
5 | Function0 | 1 | 2 | 1 | like(2) | 02 | 
6 | IfNot | 1 | 11 | 1 |  | 00 | 
7 | Column | 0 | 0 | 1 |  | 00 | 
8 | Function0 | 0 | 1 | 5 | **datetime(-1)** | 01 | 
9 | Column | 0 | 1 | 6 |  | 00 | 
10 | ResultRow | 5 | 2 | 0 |  | 00 | 
11 | Next | 0 | 3 | 0 |  | 01 | 
12 | Halt | 0 | 0 | 0 |  | 00 | 
13 | Transaction | 0 | 0 | 416 | 0 | 01 | 
14 | String8 | 0 | 2 | 0 | %2017% | 00 | 
15 | Goto | 0 | 1 | 0 |  | 00 |